internal/product/accessor_test.go

```diff
@@ -446,14 +452,29 @@ func Test_GetProductsByVendor(t *testing.T) {
 		)
 		defer c.db.Close()
 
-		c.mock.ExpectQuery(getProductsByVendorQuery).
-			WithArgs(vendorID).
+		query := getProductsByVendorQuery + " LIMIT $2 OFFSET $3"
+		c.mock.ExpectQuery(query).
+			WithArgs(vendorID, args.Limit, args.Offset).
 			WillReturnRows(expectedResult)
 
-		res, err := c.accessor.GetProductsByVendor(ctx, vendorID, spec)
+		totalRows := sqlmock.NewRows([]string{"count"}).AddRow(2)
+		countQuery := `SELECT COUNT(*) FROM product_vendor WHERE vendor_id=$1`
+		c.mock.ExpectQuery(countQuery).
+			WithArgs(vendorID).
+			WillReturnRows(totalRows)
+
+		expect := &AccessorGetProductsByVendorPaginationData{
+			Products: products,
+			Metadata: database.PaginationMetadata{
+				TotalPage:    1,
+				CurrentPage:  1,
+				TotalEntries: 2,
+			},
+		}
 
+		res, err := c.accessor.GetProductsByVendor(ctx, vendorID, spec)
 		c.g.Expect(err).To(gomega.BeNil())
-		c.g.Expect(res).To(gomega.BeComparableTo(products))
+		c.g.Expect(res).To(gomega.BeComparableTo(expect))
```

Comment:
Minor issue but I think it is better to use `c.g.Expect(res).To(gomega.BeComparableTo(expected))`